implementation/src/main/java/io/smallrye/config/AbstractLocationConfigSourceFactory.java

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    private void tryClassPath(final URI uri, final ConfigSourceContext context, final List<ConfigSource> configSources) {
+        try {
+            consumeAsPaths(uri.getPath(), new ConfigSourcePathConsumer(this, context, configSources));
+        } catch (IOException e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        } catch (IllegalArgumentException e) {
+            fallbackToUnknownProtocol(uri, configSources);
+        }
+    }
+
+    private void tryJar(final URI uri, final ConfigSourceContext context, final List<ConfigSource> configSources) {
+        try {
+            consumeAsPath(toURL(uri), new ConfigSourcePathConsumer(this, context, configSources));
+        } catch (Exception e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        }
+    }
+
+    private void fallbackToUnknownProtocol(final URI uri, final List<ConfigSource> configSources) {
+        final ClassLoader classLoader = SecuritySupport.getContextClassLoader();
+        try {
+            Enumeration<URL> resources = classLoader.getResources(uri.toString());
+            while (resources.hasMoreElements()) {
+                final URL resourceUrl = resources.nextElement();
+                if (validExtension(resourceUrl.getFile())) {
+                    addConfigSource(resourceUrl, configSources);
+                }
+            }
+        } catch (IOException e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        }
+    }
+
+    private void tryHttpResource(final URI uri, final ConfigSourceContext context, final List<ConfigSource> configSources) {
+        if (validExtension(uri.getPath())) {
+            addConfigSource(uri, configSources);
+            tryProfiles(uri, context, configSources);
+        }
+    }
+
+    private void tryProfiles(final URI uri, final ConfigSourceContext context, final List<ConfigSource> configSources) {
+        final List<String> profiles = context.getProfiles();
+        for (int i = profiles.size() - 1; i >= 0; i--) {
+            final int lastOrdinal = configSources.get(configSources.size() - 1).getOrdinal();
+            final URL profileToFileName = toURL(addProfileToFileName(uri, profiles.get(i)));
+            consumeAsPath(profileToFileName, path -> addProfileConfigSource(profileToFileName, lastOrdinal + 1, configSources));
+        }
+    }
+
+    private static URL toURL(final URI uri) {
+        try {
+            return uri.toURL();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private void addConfigSource(final URI uri, final List<ConfigSource> configSources) {
+        addConfigSource(toURL(uri), configSources);
+    }
+
+    private void addConfigSource(final URL url, final List<ConfigSource> configSources) {
+        try {
+            final ConfigSource configSource = loadConfigSource(url);
+            if (!configSource.getProperties().isEmpty()) {
```

Comment:
This might be an expensive operation since getProperties() isn't always supported, right?  I'd expect it to be *more likely* that `getPropertyNames().isEmpty()` would be performant - in the worst case it's just a view on `getProperties()`.